src/containers/PDiskPage/DecommissionButton/DecommissionButton.tsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+
+import {ChevronDown} from '@gravity-ui/icons';
+import type {DropdownMenuItem, DropdownMenuProps} from '@gravity-ui/uikit';
+import {Button, DropdownMenu, Icon, Popover} from '@gravity-ui/uikit';
+
+import {CriticalActionDialog} from '../../../components/CriticalActionDialog/CriticalActionDialog';
+import {isErrorWithRetry} from '../../../components/CriticalActionDialog/utils';
+import type {EDecommitStatus} from '../../../types/api/pdisk';
+import {wait} from '../../../utils';
+import {cn} from '../../../utils/cn';
+import {pDiskPageKeyset} from '../i18n';
+
+import decommissionIcon from '../../../assets/icons/decommission.svg';
+
+import './DecommissionButton.scss';
+
+const b = cn('ydb-pdisk-decommission-button');
+
+function getDecommissionWarningText(decommission?: EDecommitStatus) {
+    if (decommission === 'DECOMMIT_IMMINENT') {
+        return pDiskPageKeyset('decommission-dialog-imminent-warning');
+    }
+    if (decommission === 'DECOMMIT_PENDING') {
+        return pDiskPageKeyset('decommission-dialog-pending-warning');
+    }
+    if (decommission === 'DECOMMIT_REJECTED') {
+        return pDiskPageKeyset('decommission-dialog-rejected-warning');
+    }
+    if (decommission === 'DECOMMIT_NONE') {
+        return pDiskPageKeyset('decommission-dialog-none-warning');
+    }
+
+    return undefined;
+}
+
+interface DecommissionButtonProps {
+    decommission?: EDecommitStatus;
+    onConfirmAction: (newDecomissionStatus?: EDecommitStatus, isRetry?: boolean) => Promise<void>;
+    onConfirmActionSuccess: (() => Promise<void>) | VoidFunction;
+    buttonDisabled?: boolean;
+    popoverDisabled?: boolean;
+}
+
+export function DecommissionButton({
+    decommission,
+    onConfirmAction,
+    onConfirmActionSuccess,
+    buttonDisabled,
+    popoverDisabled,
+}: DecommissionButtonProps) {
+    const [newDecommission, setNewDecommission] = React.useState<EDecommitStatus | undefined>();
+    const [buttonLoading, setButtonLoading] = React.useState(false);
+    const [withRetry, setWithRetry] = React.useState(false);
+
+    const handleConfirmAction = async (isRetry?: boolean) => {
+        setButtonLoading(true);
+        await onConfirmAction(newDecommission, isRetry);
+        setButtonLoading(false);
+    };
+
+    const handleConfirmActionSuccess = async () => {
+        setWithRetry(false);
+
+        if (onConfirmActionSuccess) {
+            setButtonLoading(true);
+
+            // Decommission needs some time to change
+            // Wait for some time to send request for updated data
+            await wait(5000);
+
+            try {
+                await onConfirmActionSuccess();
+            } catch {
```

Comment:
This silently swallows any errors. It should at least log the error or notify the user.

Reply:
It supposed, that data refetch errors will be handled in the component, where data is used. Example:

![Screenshot 2024-09-03 at 16 56 55](https://github.com/user-attachments/assets/b20e5775-b905-45e8-81c5-6a6fd05ff618)